src/brogue/MainMenu.c

```diff
@@ -260,136 +279,336 @@ void initializeMenuFlames(boolean includeTitle,
 
     // Empty nextGamePath and nextGameSeed so that the buttons don't try to load an old game path or seed.
     rogue.nextGamePath[0] = '\0';
     rogue.nextGameSeed = 0;
 
-    // Initialize the title menu buttons.
-    encodeMessageColor(whiteColorEscape, 0, &white);
-    encodeMessageColor(goldColorEscape, 0, KEYBOARD_LABELS ? &itemMessageColor : &white);
-    sprintf(newGameText, "      %sN%sew Game      ", goldColorEscape, whiteColorEscape);
-    sprintf(customNewGameText, " %sN%sew Game (custom) ", goldColorEscape, whiteColorEscape);
-    b = 0;
-    button = -1;
-
-    initializeButton(&(buttons[b]));
-    strcpy(buttons[b].text, newGameText);
-    buttons[b].hotkey[0] = 'n';
-    buttons[b].hotkey[1] = 'N';
-    b++;
-
-    initializeButton(&(buttons[b]));
-    sprintf(buttons[b].text, "     %sO%spen Game      ", goldColorEscape, whiteColorEscape);
-    buttons[b].hotkey[0] = 'o';
-    buttons[b].hotkey[1] = 'O';
-    b++;
-
-    initializeButton(&(buttons[b]));
-    sprintf(buttons[b].text, "   %sV%siew Recording   ", goldColorEscape, whiteColorEscape);
-    buttons[b].hotkey[0] = 'v';
-    buttons[b].hotkey[1] = 'V';
-    b++;
-
-    initializeButton(&(buttons[b]));
-    sprintf(buttons[b].text, "    %sH%sigh Scores     ", goldColorEscape, whiteColorEscape);
-    buttons[b].hotkey[0] = 'h';
-    buttons[b].hotkey[1] = 'H';
-    b++;
-
-    initializeButton(&(buttons[b]));
-    sprintf(buttons[b].text, "        %sQ%suit        ", goldColorEscape, whiteColorEscape);
-    buttons[b].hotkey[0] = 'q';
-    buttons[b].hotkey[1] = 'Q';
-    b++;
-
-    x = COLS - 1 - 20 - 2;
-    y = ROWS - 1;
-    for (i = b-1; i >= 0; i--) {
-        y -= 2;
-        buttons[i].x = x;
-        buttons[i].y = y;
-        buttons[i].buttonColor = titleButtonColor;
-        buttons[i].flags |= B_WIDE_CLICK_AREA;
-    }
-
     blackOutScreen();
-    clearDisplayBuffer(shadowBuf);
-    initializeButtonState(&state, buttons, b, x, y, 20, b*2-1);
-    rectangularShading(x, y, 20, b*2-1, &black, INTERFACE_OPACITY, shadowBuf);
-    drawButtonsInState(&state);
 
+    // Initialize the main menu with buttons stacked on top of the quit button
+    windowpos quitButtonPosition = {COLS - 20, ROWS - 3};
+    initializeMainMenu(&mainMenu, mainButtons, quitButtonPosition, mainShadowBuf);
+
+    // Display the title and flames
     initializeMenuFlames(true, colors, colorStorage, colorSources, flames, mask);
     rogue.creaturesWillFlashThisTurn = false; // total unconscionable hack
 
+    rogueEvent theEvent;
+    short mainIndex = -1;
+    short flyoutIndex = -1;
+    windowpos bPos;
+
+    // Outer loop for menu navigation (showing/hiding the flyout menus). Repeat until the user selects
+    // a button with a command handled by mainBrogueJunction.
+    do {
+        if (isFlyoutActive()) {
+            bPos = getNextGameButtonPos(mainButtons);
+            initializeFlyoutMenu(&flyoutMenu, flyoutShadowBuf, flyoutButtons, (windowpos){FLYOUT_X, bPos.window_y});
+    }
```

Comment:
Indentation, basically please review indentation here, it's hard to read 😿

Reply:
Yikes! So I think the indentation issues are an artifact of the way I cleaned up the commit history. I started with a messy but working branch and applied successive changes to a new PR branch.